### DISCUSSION TITLE
nullish coalescing operator is not transformed in my svelte + rollup project

### DISCUSSION THREAD

**Discussion by cascading-jox** [babel/babel#14704] [Q&A] [answered]
Hi! I can't seem to get rollup + babel to transform a nullish coalescing operator to es5 syntax. The same code works when building with vite so it might be related to the plugin. Is something wrong with my rollup/babel config? I would really appreciate if anyone could take a quick look!

Minimal repo with svelte: https://github.com/cascading-jox/babel-rollup-nullish-coalescing

(the nullish coalescing operator is still present in the public/build/bundle.js:346 after running `npm run build`)

**Comment by JLHwung** [comment, marked as answer]
Thanks for the reproduction repo, it is very helpful. After the svelte compilation, the rollup-plugin-svelte will NOT change the module extension from `.svelte` to `.js`, so you need to tell rollup-plugin-babel that the `.svelte` file (compiled by svelte) should be handled by Babel, too:

```js
babel({
  extensions: [".svelte", ".js" /* or other extensions */],
  // ...other rollup-plugin-babel options
  babelHelpers: 'bundled',
})
```

**Reply by cascading-jox** [reply]
Thank you so much!! I had one of those programming moments where it seemed everything was correct but it still didn't work and you start feeling like you are going crazy. How silly of me to forget about the extension... Thanks again!

**Reply by JLHwung** [reply]
You are welcome. Yeah the config looked good to me, too. My first try is to see whether `??` comes from node modules (common scenarios) but it's not. Then I checked the source of rollup-plugin-svelte and rollup-plugin-babel and realized that it is a rollup plugin integration issue.
